src/bapsicle.ts

```diff
@@ -0,0 +1,170 @@
+              }
+            } else {
+              store.dispatch(MixerState.load(channel, null));
+            }
+
+            if (
+              !bapsicle_state.initialised ||
+              (bapsicle_state.loaded_item && !bapsicle_state.loaded)
+            ) {
+              store.dispatch(MixerState.itemLoadError({ player: channel }));
+            }
+
+            store.dispatch(
+              MixerState.setTimeLength({
+                player: channel,
+                time: bapsicle_state.length,
+              })
+            );
+            store.dispatch(
+              MixerState.setTimeCurrent({
+                player: channel,
+                time: bapsicle_state.pos_true,
+              })
+            );
+            store.dispatch(
+              MixerState.setAutoAdvance({
+                player: channel,
+                enabled: bapsicle_state.auto_advance,
+              })
+            );
+            store.dispatch(
+              MixerState.setPlayOnLoad({
+                player: channel,
+                enabled: bapsicle_state.play_on_load,
+              })
+            );
+            store.dispatch(
+              MixerState.setRepeat({
+                player: channel,
+                mode: bapsicle_state.repeat.toLowerCase(),
+              })
+            );
+
+            if (!("show_plan" in message.data)) {
+              console.error("Show plan data missing from status");
+              console.error(message.data);
+            }
+            // Update the list of plan items
+            var raw_planitems: [any] = message.data.show_plan;
+            var planItems: Array<PlanItem> = [];
+            raw_planitems.forEach((raw_planitem) => {
+              var planItem: PlanItem = {
+                timeslotitemid: String(raw_planitem.timeslotItemId),
+                channel: message.channel,
+                played: false,
+                playedAt: raw_planitem.played_at,
+                playCount: raw_planitem.play_count,
+                type: raw_planitem.trackId ? "central" : "aux",
+                trackid: raw_planitem.trackId,
+                managedid: raw_planitem.managedId,
+                auxid: raw_planitem.managedId,
+                ...raw_planitem,
+              };
+              planItem = planItem as PlanItem;
+              planItems.push(planItem);
+            });
+            store.dispatch(
+              getShowplanSuccessChannel({
+                channel: message.channel,
+                planItems: planItems,
+              })
+            );
+        }
+      } else if ("message" in message) {
+        if (message.message === "Hello") {
+          //store.dispatch(
+          //  connection.actions.setServerState(message.serverName)
+          //);
+        }
+      } else {
+        console.log("Unhandled: ", message);
+      }
+    };
+  }
+  return next(action);
+};
+
+export function sendBAPSicleChannel(message: any): void {
+  if (BAPSicleWS) {
```

Comment:
This will drop messages if this is called before the connection is established. I imagine it's fairly unlikely that this will happen, but it may be worth sticking in a `console.warn` - if nothing else, it'll make debugging easier if it does end up happening.